content/panorama/scripts/team_select.js

```diff
@@ -30,6 +30,7 @@ function hidePregameUI () {
 function showPregameUI () {
   FindDotaHudElement('PreGame').style.opacity = 1;
   FindDotaHudElement('PreGame').style.visibility = 'visible';
+  FindDotaHudElement('PreGame').FindChildTraverse( 'GameModeLabel' ).text = "Open Angel Arena";
```

Comment:
It's just a KV file in `game/panorama/localization`. Put in a value for "dota_gamemode_15" in addon_english.txt or other relevant language. You can still create an issue for the problem of the game length result not being shown though.